cmd/pullrequest-init/github.go

```diff
@@ -0,0 +1,227 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"golang.org/x/oauth2"
+
+	"github.com/google/go-github/github"
+	"go.uber.org/zap"
+)
+
+// GitHubHandler handles interactions with the GitHub API.
+type GitHubHandler struct {
+	*github.Client
+
+	owner, repo string
+	prNum       int
+
+	Logger *zap.SugaredLogger
+}
+
+// NewGitHubHandler initializes a new handler for interacting with GitHub
+// resources.
+func NewGitHubHandler(ctx context.Context, logger *zap.SugaredLogger, rawURL string) (*GitHubHandler, error) {
+	token := strings.TrimSpace(os.Getenv("GITHUBOAUTHTOKEN"))
```

Comment:
hm where does `GITHUBOAUTHTOKEN` come from? maybe it could be mentioned in the README? (what will be setting this?)

Reply:
It's passed in as an environment variable from the generated Container spec. Added a note in the README.